tests/ga/test_extension.py

```diff
@@ -1961,5 +2026,174 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
+
+                self.assertEqual(0, patch_set_env.call_count, "No Env should be set for disable failures")
+
+            # When Uninstall Fails
+            with patch.object(ExtHandlerInstance, "uninstall", side_effect=ExtensionError("Uninstall Failed")):
+                with self.assertRaises(ExtensionUpdateError):
+                    ExtHandlersHandler._update_extension_handler(old_handler_i, new_handler_i)
+
+                self.assertEqual(0, patch_set_env.call_count, "No Env should be set for uninstall failures")
+
+    def test_set_env_variable_should_set_handler_env_variable(self, *args):
+        handler_i = self._get_ext_handler_instance('foo', '1.0.0')
+
+        name = "test_variable"
+        another_var = "another_variable"
+        initial_value = "test_value"
+        changed_value = "new_test_value"
+
+        handler_i.set_handler_env_variable({name: initial_value})
+        handler_i.set_handler_env_variable({another_var: changed_value})
+        self.assertTrue(name in handler_i.handler_env and another_var in handler_i.handler_env,
+                        "The variables not there in handler_environment")
+        self.assertEqual(initial_value, handler_i.handler_env[name], "The value for env variable doesn't match")
+        self.assertEqual(changed_value, handler_i.handler_env[another_var], "The value for env variable doesn't match")
+
+        # Changing the value with the same name should update the value
+        handler_i.set_handler_env_variable({name: changed_value})
+        self.assertEqual(changed_value, handler_i.handler_env[name], "The new value is not set")
+
+    def test_remove_key_from_env_should_remove_handler_env_variable(self, *args):
+        handler_i = self._get_ext_handler_instance('foo', '1.0.0')
+
+        name_to_delete = "test_variable"
+        value_to_delete = "test_value"
+        name_to_keep = "keep_variable"
+        value_to_keep = "keep_value"
+        handler_i.set_handler_env_variable({name_to_delete: value_to_delete})
+        handler_i.set_handler_env_variable({name_to_keep: value_to_keep})
+
+        self.assertTrue(name_to_delete in handler_i.handler_env and name_to_keep in handler_i.handler_env,
+                        "The variable not there in handler_environments")
+        self.assertTrue(value_to_delete == handler_i.handler_env[name_to_delete] and
+                        value_to_keep == handler_i.handler_env[name_to_keep],
+                        "The value for env variable doesn't match")
+
+        handler_i.remove_key_from_handler_env(name_to_delete)
+        self.assertTrue(name_to_delete not in handler_i.handler_env, "The variable not removed from handler_env list")
+
+        # Retrying the removal operation to ensure that no exception is thrown if key not found
+        try:
+            handler_i.remove_key_from_handler_env(name_to_delete)
+        except:
+            self.fail("Removing a key that's not present shouldn't throw an error")
+
+
```

Comment:
we should also have tests that check the env variables in the actual extension commands

Reply:
Done